File: java/src/main/java/com/function/GetOrders.java
package com.function;

import java.util.List;
import java.util.Optional;

import com.function.util.Order;
import com.microsoft.azure.functions.ExecutionContext;
import com.microsoft.azure.functions.HttpMethod;
import com.microsoft.azure.functions.HttpRequestMessage;
import com.microsoft.azure.functions.HttpResponseMessage;
import com.microsoft.azure.functions.HttpStatus;
import com.microsoft.azure.functions.annotation.AuthorizationLevel;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.annotation.HttpTrigger;

public class GetOrders {

    @FunctionName("GetOrders")
    public HttpResponseMessage run(
            @HttpTrigger(name = "req", 
                methods = {HttpMethod.GET}, 
                authLevel = AuthorizationLevel.FUNCTION,
                route = "orders") 
            HttpRequestMessage<Optional<String>> request,
            final ExecutionContext context) {
        
        try
        {
            List<Object> result = null;
            final Boolean autoDecrypt = Boolean.parseBoolean(request.getQueryParameters().get("autoDecrypt"));

            //If filtered by customerName (encrypted) requires client encryption
            if (request.getQueryParameters().containsKey("customerName"))
                result = Order.FindOrder("customerName", request.getQueryParameters().get("customerName"), true, autoDecrypt);
            else if (request.getQueryParameters().containsKey("customerId"))
                result = Order.FindOrder("customerId", Integer.parseInt(request.getQueryParameters().get("customerId")), false, autoDecrypt);
            else
                return request.createResponseBuilder(HttpStatus.BAD_REQUEST).body("customerName or customerId query parameters required!").build();

            if (result == null)
                return request.createResponseBuilder(HttpStatus.NOT_FOUND).build();
            else
                return request.createResponseBuilder(HttpStatus.OK).body(result).build();
        }
        catch (Exception ex)
        {
            context.getLogger().warning(ex.getMessage());
            return request.createResponseBuilder(HttpStatus.BAD_REQUEST).body(ex.getMessage()).build();
        }
    }
}

File: dotnet/src/Functions/GetOrders.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.Extensions.Logging;
using MongoEncryption.Util;
using System;
using System.Threading.Tasks;

namespace MongoEncryption.Functions
{
    public static class GetOrders
    {
        [FunctionName("GetOrders")]
        public static async Task<IActionResult> RunAsync(
            [HttpTrigger(AuthorizationLevel.Function, 
            "get", 
            Route = "orders")] HttpRequest req,
            ILogger log)
        {
            try
            {
                object result = null;
                bool autoDecrypt = true;

                if (req.Query.ContainsKey("autoDecrypt"))
                    autoDecrypt = bool.Parse(req.Query["autoDecrypt"]);

                //If filtered by customerName (encrypted) requires client encryption
                if (req.Query.ContainsKey("customerName"))
                    result = await Order.FindOrderAsync("customerName", req.Query["customerName"].ToString(), true, autoDecrypt);
                else if (req.Query.ContainsKey("customerId"))
                    result = await Order.FindOrderAsync("customerId", int.Parse(req.Query["customerId"]), false, autoDecrypt);
                else
                    return new BadRequestObjectResult("customerName or customerId query parameters required!");

                if (result == null)
                    return new NotFoundResult();
                else
                    return new OkObjectResult(result);
            }
            catch (Exception ex)
            {
                log.LogError(ex, ex.Message);
                return new BadRequestObjectResult(ex.Message);
            }
        }
    }
}
